code-challenges.test.js

```diff
@@ -25,27 +27,154 @@ const colors2 = [
   "aquamarine",
   "saffron"
 ]
+
+it("takes in an array, removes the first item from the array and shuffles the remaining content", () => {
+      expect(isherYates(colors1)).toEqual(expect.arrayContaining(["yellow", "blue", "pink", "green"]))
+      expect(isherYates(colors1)).not.toEqual(expect.arrayContaining(["purple"]))
+      expect(isherYates(colors2)).toEqual(expect.arrayContaining(["indigo", "periwinkle", "ochre", "aquamarine", "saffron"]))
+      expect(isherYates(colors2)).not.toEqual(expect.arrayContaining(["chartreuse"]))
+})
+})
```

Comment:
Really nice work!  I like that you are also verifying that the first item is actually removed in your tests.  Initially I was going to comment on the different names for the function, but I kept scrolling and saw you created a second function called isherYates.  It would be cool to create a test for both and check the efficiency in your tests of one approach versus the other if you wanted to stretch this even further.  Excellent job on this problem though!  Love the thorough research and that you tried multiple approaches!

Reply:
Aaron gave me the idea for checking that the first element was removed while I was listening in on the main channel on Zoom.